models/base/scratch/bigquery/snowplow_mobile_base_events_this_run.sql

```diff
@@ -66,13 +61,13 @@ with events as (
   inner join {{ ref('snowplow_mobile_base_sessions_this_run') }} as b
   on {{ session_id }} = b.session_id
 
-  where {{ snowplow_utils.timestamp_diff('a.collector_tstamp', 'b.start_tstamp', 'day')}} <= {{ var("snowplow__max_session_days", 3) }}
-  and {{ snowplow_utils.timestamp_diff('a.dvce_created_tstamp', 'a.dvce_sent_tstamp', 'day') }} <= {{ var("snowplow__days_late_allowed", 3) }}
+  where {{ snowplow_utils.timestamp_add('day', var("snowplow__max_session_days", 3), 'a.collector_tstamp') }} <= b.start_tstamp
+  and {{ snowplow_utils.timestamp_add('day', var("snowplow__days_late_allowed", 3), 'a.dvce_created_tstamp') }} <= a.dvce_sent_tstamp
   and a.collector_tstamp >= {{ lower_limit }}
   and a.collector_tstamp <= {{ upper_limit }}
   {% if var('snowplow__derived_tstamp_partitioned', true) and target.type == 'bigquery' | as_bool() %} -- BQ only
-    and cast(a.derived_tstamp as datetime) >= {{ dbt_utils.dateadd('hour', -1 * var('snowplow__lookback_window_hours', 6), lower_limit) }}
-    and cast(a.derived_tstamp as datetime) <= {{ dbt_utils.dateadd('hour', var('snowplow__lookback_window_hours', 6), upper_limit) }}
+    and a.derived_tstamp >= {{ snowplow_utils.timestamp_add('hour', -1, lower_limit) }}
+    and a.derived_tstamp <= {{ snowplow_utils.timestamp_add('hour', 1, upper_limit) }}
```

Comment:
Not we need to expand the upper limit as the `collector_tstamp` should always be greater than the `derived_tstamp`